pkg/repo/resolver/scripts/prepare-base.sh.tpl

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+set -euo pipefail
+
+#  Template Fields
+#  WorkDir     - directory from where this script will be running
+#  ImgPath     - path to the image that will be prepared
+#  ImgType     - type of the image (either .iso, or .raw)
+#  ArchiveName - name of the virtual disk archive that will be created from this image
+
+WORK_DIR={{.WorkDir}}
+IMG_PATH={{.ImgPath}}
+
+{{ if eq .ImgType "iso" -}}
+xorriso -osirrox on -indev $IMG_PATH extract / $WORK_DIR/iso-root/
+
+ISO_ROOT=$WORK_DIR/iso-root/
+cd $ISO_ROOT
+
+ISO_SQUASHFS=`find $ISO_ROOT -name "*.squashfs"`
+unsquashfs $ISO_SQUASHFS 
```

Comment:
I'm playing with unsquash now for other stuff and there's a `-d` that lets you explicitly indicate where to extract to. The default is currently `squashfs-root` which is hardcoded on the next line, but it would be safer to explicitly pass in the destination with `-d` in case the default behavior changes.

Reply:
Yea, explicitly specifying the dir name would be better.